src/instructlab/configuration.py

```diff
@@ -619,7 +647,228 @@ def ensure_storage_directories_exist():
+            lora_rank=0,
+            max_batch_len=30000,
+            max_seq_len=4096,
+            save_samples=1000,
+            deepspeed_cpu_offload_optimizer=False,
+            is_padding_free=True,
+            num_epochs=7,
+            model_path=os.path.join(
+                DEFAULTS._cache_home, "models/instructlab/granite-7b-lab"
+            ),
+        )
+
+        FOUR_L_FORTY_GPU = _train(
+            additional_args={
+                "warmup_steps": 10,
+                "learning_rate": 2e-5,
+                "lora_dropout": 0.1,
+                "lora_alpha": 32,
+                "deepspeed_cpu_offload_optimizer_ratio": 1,
+                "deepspeed_cpu_offload_optimizer_pin_memory": False,
+            },
+            ckpt_output_dir=os.path.join(DEFAULTS._data_dir, "checkpoints"),
+            data_output_dir=os.path.join(DEFAULTS._data_dir, "internal"),
+            data_path=os.path.join(DEFAULTS._data_dir, "datasets"),
+            nproc_per_node=4,
+            effective_batch_size=96,
+            lora_quantize_dtype=None,
+            lora_rank=0,
+            max_batch_len=30000,
+            max_seq_len=4096,
+            save_samples=1000,
+            num_epochs=7,
+            deepspeed_cpu_offload_optimizer=True,
+            is_padding_free=True,
+            model_path=os.path.join(
+                DEFAULTS._cache_home, "models/instructlab/granite-7b-lab"
+            ),
+        )
+
+        EIGHT_L_FOUR_GPU = _train(
+            additional_args={
+                "warmup_steps": 10,
+                "learning_rate": 2e-5,
+                "lora_dropout": 0.1,
+                "lora_alpha": 32,
+                "deepspeed_cpu_offload_optimizer_ratio": 1,
+                "deepspeed_cpu_offload_optimizer_pin_memory": False,
+            },
+            ckpt_output_dir=os.path.join(DEFAULTS._data_dir, "checkpoints"),
+            data_output_dir=os.path.join(DEFAULTS._data_dir, "internal"),
+            data_path=os.path.join(DEFAULTS._data_dir, "datasets"),
+            nproc_per_node=4,
+            effective_batch_size=96,
+            lora_quantize_dtype=None,
+            lora_rank=0,
+            max_batch_len=30000,
+            max_seq_len=4096,
+            save_samples=1000,
+            num_epochs=7,
+            deepspeed_cpu_offload_optimizer=True,
+            is_padding_free=True,
+            model_path=os.path.join(
+                DEFAULTS._cache_home, "models/instructlab/granite-7b-lab"
+            ),
+        )
+
+        to_write = {
+            DEFAULTS.TRAIN_A100_H100_X4_PROFILE: FOUR_GPU_TRAIN_AH,
+            DEFAULTS.TRAIN_A100_H100_X8_PROFILE: EIGHT_GPU_TRAIN_AH,
+            DEFAULTS.TRAIN_A100_H100_X2_PROFILE: TWO_GPU_TRAIN_AH,
+            DEFAULTS.TRAIN_L40_X8_PROFILE: EIGHT_L_FORTY_GPU,
+            DEFAULTS.TRAIN_L40_X4_PROFILE: FOUR_L_FORTY_GPU,
+            DEFAULTS.TRAIN_L4_X8_PROFILE: EIGHT_L_FOUR_GPU,
+        }
+
+        for file, train_cfg in to_write.items():
```

Comment:
nit: this section seems identical to how we dump above for the 'custom profile dir' scenario. It would be nice to have this deduplicated.